core/src/main/scala/org/apache/spark/package.scala

```diff
@@ -41,7 +41,46 @@ package org.apache
  * level interfaces. These are subject to changes or removal in minor releases.
  */
 
+import java.util.Properties
+
 package object spark {
-  // For package docs only
-  val SPARK_VERSION = "2.0.0-SNAPSHOT"
+
+  private object SparkBuildInfo {
+    val unknownProp = "<unknown>"
+    val props = new Properties()
+    val resourceStream = Thread.currentThread().getContextClassLoader.
+      getResourceAsStream("spark-version-info.properties")
+    val (spark_version: String, spark_branch: String, spark_revision: String,
+      spark_build_user: String, spark_repo_url: String, spark_build_date: String) =
```

Comment:
Could you place the temp `val`s inside a scoped block here, so they can be garbage collected? e.g.:

```
val (spark_version: String, spark_branch: String, spark_revision: String,
  spark_build_user: String, spark_repo_url: String, spark_build_date: String) = {

  val unknownProp = "<unknown>"
  ...
  try {
    ...
  } 
}
```

Also, when you have multiple "parameters", we generally do the following:

```
val (
    spark_version: String, 
    spark_branch: String, 
    spark_revision: String,
    spark_build_user: String,
    spark_repo_url: String,
    spark_build_date: String) = {
  // code
}
```